primitives/npos-elections/src/balancing.rs

```diff
@@ -90,8 +90,7 @@ pub(crate) fn balance_voter<AccountId: IdentifierT>(
 		.filter(|e| e.candidate.borrow().elected)
 		.collect::<Vec<&mut Edge<AccountId>>>();
 
-	// Either empty, or a self vote. Not much to do in either case.
-	if elected_edges.len() <= 1 {
+	if elected_edges.len() == 0 {
```

Comment:
why does this have to change?

Reply:
if the voter is voting for just one edge, then there is nothing to balance further and we should return. Previous code seems correct to me.

Reply:
This is definitely the most contentious change and I am not sure about it. However, even if there is no rebalancing to be done, this method actually populates some fields (namely `edge.weight` and `backed_stake` of the candidate).
I believe that `phragMMS` fills those using the other helper methods.
Doing this change here seemed to make the `MMS` algorithm work better. Happy to take suggestions though. The alternative is to keep the code as is and have an initialiser in the `mms` implementation, but it does not seem to work as well.

Reply:
I have reverted this.